pyvista/utilities/geometric_objects.py

```diff
@@ -1264,9 +1268,82 @@ def Rectangle(points=None):
     >>> pointa = [1.0, 0.0, 0.0]
     >>> pointb = [1.0, 1.0, 0.0]
     >>> pointc = [0.0, 1.0, 0.0]
-    >>> pointd = [0.0, 0.0, 0.0]
-    >>> rectangle = pyvista.Rectangle([pointa, pointb, pointc, pointd])
+    >>> rectangle = pyvista.Rectangle([pointa, pointb, pointc])
     >>> rectangle.plot(show_edges=True, line_width=5)
+    """
+    if points is None:
+        points = [[1.0, 0.0, 0.0], [1.0, 1.0, 0.0], [0.0, 1.0, 0.0]]
+    if len(points) == 4:
+        warnings.warn(
+            'Rectangle defined by 4 points is deprecated. Please use ``pyvista.Quadrilateral``.',
+            PyVistaDeprecationWarning,
+        )
+        return Quadrilateral(points)
+    elif len(points) != 3:
+        raise TypeError('Points must be given as length 3 np.ndarray or list')
+
+    check_valid_vector(points[0], 'points[0]')
+    check_valid_vector(points[1], 'points[1]')
+    check_valid_vector(points[2], 'points[2]')
+
+    point_0 = np.array(points[0])
+    point_1 = np.array(points[1])
+    point_2 = np.array(points[2])
+
+    vec_01 = point_1 - point_0
+    vec_02 = point_2 - point_0
+    vec_12 = point_2 - point_1
+
+    scalar_pdct_01_02 = np.dot(vec_01, vec_02)
+    scalar_pdct_01_12 = np.dot(vec_01, vec_12)
+    scalar_pdct_02_12 = np.dot(vec_02, vec_12)
+
+    non_null_scalar_products = [
```

Comment:
Is this actually `null_scalar_products`?

Reply:
You are totally right.